lib/src/util/leak_detector_visitor.dart

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.util.leak_detector_visitor;
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:linter/src/linter.dart';
+import 'package:linter/src/util/dart_type_utilities.dart';
+
+typedef bool DartTypePredicate(DartType type);
+typedef void _VisitVariableDeclaration(VariableDeclaration node);
+typedef bool _Predicate(AstNode node);
+typedef _Predicate _PredicateBuilder(VariableDeclaration v);
+
+abstract class LeakDetectorVisitor extends SimpleAstVisitor {
+  static _PredicateBuilder _isSinkReturn =
```

Comment:
Perhaps generalize the name, given that this is now being used to detect the return of something more than a Sink.

Also, IMO this and the other fields in this class would be more readable if they were static methods rather than static function-valued variables:

```
static _Predicate _isSinkReturn(VariableDeclaration v) => ...
```